### PR TITLE
Add support for Ubuntu Lunar release

### PR DESCRIPTION
**Pull request by peterbarker**
We've had to move to venv to here - polluting the system Python is now very difficult to do.

Also incorporates a change to waf required for Lunar's Python to function for our `waf`.

### CODE REVIEW

**Review by khancyr — APPROVED** [review]
look good.
I will try to test if before the dev call.
could you add it to https://github.com/ArduPilot/ardupilot/blob/master/.github/workflows/test_environment.yml ?

**Comment by peterbarker** [comment]
> look good. I will try to test if before the dev call. could you add it to https://github.com/ArduPilot/ardupilot/blob/master/.github/workflows/test_environment.yml ?

Added.

**Comment by khancyr** [comment]
`Tools/environment_install/install-prereqs-ubuntu.sh: line 340: /github/home/venv-ardupilot/bin/activate: No such file or directory`

https://github.com/khancyr/ardupilot/actions/runs/4865329695/jobs/8675496004

Looks like venv config doesn't works in all case

File: Tools/environment_install/install-prereqs-ubuntu.sh [hunk 1]
```diff
@@ -288,12 +314,53 @@ fi
...
+    echo "Inside docker, we add the tools path into .bashrc directly"
+    SHELL_LOGIN=".ardupilot_env"
+    echo "# ArduPilot env file. Need to be loaded by your Shell." > ~/$SHELL_LOGIN
+fi
+
+PIP_USER_ARGUMENT="--user"
+# create a Python venv on more recent releases:
+if [ ${RELEASE_CODENAME} == 'lunar' ]; then
+    $APT_GET install python3.11-venv
+    python3 -m venv venv-ardupilot
```
Comment (khancyr) [inline Tools/environment_install/install-prereqs-ubuntu.sh:336]: python3 -m venv "$HOME/venv-ardupilot"

**Review by khancyr — CHANGES_REQUESTED** [review]

File: Tools/environment_install/install-prereqs-ubuntu.sh (hunk 1 above)
Comment (khancyr) [reply Tools/environment_install/install-prereqs-ubuntu.sh:336]: and after it works !

Comment (peterbarker) [reply Tools/environment_install/install-prereqs-ubuntu.sh:336]: > Could we place it inside the repo and add it to the ignore?

That would be bad for users using Vagrant to bring up lots of different sorts of machines.

Comment (peterbarker) [reply Tools/environment_install/install-prereqs-ubuntu.sh]: > and after it works !

Excellent!  I've made that change.

**Review by tridge — APPROVED** [review]

**Comment by Hwurzburg** [comment]
@peterbarker what wiki is needed? this is just the new enviro install script....

**Comment by khancyr** [comment]
probably that starting from ubuntu 23.04 we start using python venv . That something to underline I think

**Comment by Hwurzburg** [comment]
so what impact is that to a user setting up the envir or running a build? its invisible to them, or do they need to change how to run build commands,etc.

**Pull request merged**